### PR TITLE
Picking up firebase env from .env file

### PR DESCRIPTION
**Pull request by vineetshar**
- [x] Tested on local & docker
- [ ] Documentation Updated
- [ ] Attached required context / screenshot

### CODE REVIEW

**Review by cash1998 — APPROVED** [review]

File: server/firebase_setup.py [hunk 1]
```diff
@@ -1,11 +1,29 @@
...
+import json
+import logging
 
 def firebase_init():
-    cred = credentials.Certificate(
-        f"{os.getcwd()}/firebase_service_account.json"
-    )
-    firebase_admin.initialize_app(cred)
+    service_account_base64 = os.getenv('FIREBASE_SERVICE_ACCOUNT')
+    print("Loading service account")
```
Comment (dhirenmathur) [inline server/firebase_setup.py:10]: Remove print statement

File: server/firebase_setup.py [hunk 2]
```diff
@@ -1,11 +1,29 @@
...
+    if service_account_base64:
+        try:
+            service_account_info = base64.b64decode(service_account_base64).decode('utf-8')
+            service_account_json = json.loads(service_account_info)
+            cred = credentials.Certificate(service_account_json)
+            logging.info("Loaded Firebase credentials from environment variable.")
+            print("Loaded Firebase credentials from environment variable.")
```
Comment (dhirenmathur) [inline server/firebase_setup.py:17]: Print statement

File: server/firebase_setup.py [hunk 3]
```diff
@@ -1,11 +1,29 @@
...
+        except Exception as e:
+            logging.info(f"Error decoding Firebase service account from environment variable: {e}")
+            cred = credentials.Certificate(f"{os.getcwd()}/firebase_service_account.json")
+            logging.info("Loaded Firebase credentials from local file as fallback.")
+            print("Loaded Firebase credentials from local file as fallback.")
```
Comment (dhirenmathur) [inline server/firebase_setup.py:22]: Print

File: server/firebase_setup.py [hunk 4]
```diff
@@ -1,11 +1,29 @@
...
+    else:
+        cred = credentials.Certificate(f"{os.getcwd()}/firebase_service_account.json")
+        logging.info("Loaded Firebase credentials from local file.")
+        print("Loaded Firebase credentials from local file.")
```
Comment (dhirenmathur) [inline server/firebase_setup.py:26]: One more

**Review by dhirenmathur — CHANGES_REQUESTED** [review]

**Comment by Harigithub11** [comment]
This pull request was merged before a formal code review and approval process was established. The change was reviewed after the merge and accepted as a known exception. The repository now enforces pull request approvals through branch protection rules to prevent this from happening in the future.

**Pull request merged**